grype/presenter/cyclonedx/vuln-extension.go

```diff
@@ -0,0 +1,73 @@
+package cyclonedx
+
+import (
+	"encoding/xml"
+	"time"
+
+	"github.com/anchore/grype/internal"
+	"github.com/anchore/grype/internal/version"
+	syftCDX "github.com/anchore/syft/syft/presenter/cyclonedx"
+)
+
+// Source: https://cyclonedx.org/ext/vulnerability/
+
+// Source is the origin of the vulnerability, like Github Advisories or NVD, along
+// with a URL constructed with the vulnerability ID
+type Source struct {
+	Name string `xml:"v:name,attr"`
+	URL  string `xml:"v:url"`
+}
+
+// Raiting has information about the intensity of a vulnerability
+type Raiting struct {
```

Comment:
Should this be `Rating`? (spelling)

Reply:
(ditto comment for all other references to this and related property/variable names)

Reply:
indeed, in order to catch all out-of-spec issues like this it would be ideal to add testing against the schema in the same/similar way done in syft https://github.com/anchore/syft/pull/163 . The extra addition would be to add the [vulnerability extension XSD](https://github.com/CycloneDX/specification/blob/master/schema/ext/vulnerability-1.0.xsd) to test against.